src/v3/plugins/github/edges.js

```diff
@@ -0,0 +1,216 @@
+// @flow
+
+import {
+  type Edge,
+  type EdgeAddressT,
+  type NodeAddressT,
+  EdgeAddress,
+  NodeAddress,
+} from "../../core/graph";
+import * as GithubNode from "./nodes";
+
+export opaque type RawAddress: EdgeAddressT = EdgeAddressT;
+
+// This particular union of types is used frequently. It seems
+// reasonable that it semantically represents some kind of "post" that
+// has a parent, was authored, and can be referred to. But if one of
+// those axes of symmetry breaks down, then we shouldn't feel hesitant
+// to inline this type in the appropriate places.
+export type PostAddress =
```

Comment:
This should be exported from nodes, not edges.

Reply:
Well… as noted in the comment, it’s really not clear what, if anything,
this union type represents. It’s quite plausible that it is a false
abstraction born out of coincidence.

What do you think about splitting this up into a larger set of unions,
each of which actually does have reasonable semantics? It seems that we
might want `ChildAddress`, `ParentAddress`, `AuthorableAddress`,
`ReferentAddress`, and `TextContentAddress`. Currently, four of those
types happen to be a union of the same four members (comment, issue,
pull request, review). If there is evidence that some of these
relationships are fundamental—e.g., “everything that is authorable has
text content” (dubious)—then we can unify the appropriate unions, but if
there isn’t then we probably shouldn’t just say “eh, they’re all some
kind of ‘post’ ” and leave it at that.

With these types, we would have:

```js
export type AuthorsAddress = {|
  +type: "AUTHORS",
  +author: GithubNode.UserlikeAddress,
  +content: GithubNode.AuthorableAddress,
|};
export type MergedAsAddress = {|
  +type: "MERGED_AS",
  +pull: GithubNode.PullAddress,
|};
export type HasParentAddress = {|
  +type: "HAS_PARENT",
  +child: GithubNode.ChildAddress,
|};
export type ReferencesAddress = {|
  +type: "REFERENCES",
  +referrer: GithubNode.TextContentAddress,
  +referent: GithubNode.ReferentAddress,
|};
```

If we effect this change, then it does make sense to me to export these
types from `nodes.js` instead of `edges.js`. Otherwise, I’m not so sure.

Reply:
Discussed more: https://discordapp.com/channels/453243919774253079/454007907663740939?jump=456564101183832064

I will wait for #387 to land.